### PR TITLE
E2E test Istio access logs

### PR DESCRIPTION
**Pull request by rakesh-garimella**
## Description

Changes proposed in this pull request:

- Test Istio access logs
- Deploy the metrics app and use the metrics endpoint to trigger the get call

## Related Issues and Documents

Closes:
https://github.com/kyma-project/telemetry-manager/issues/288

## Traceability

- [x] The PR is linked to a GitHub Issue.
- [ ] New features have a milestone label set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [ ] Yes.
- [ ] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [ ] Yes.
- [ ] No, because e2e-tests are not needed.
- [ ] No, because of ...

Tests conducted for the PR:

## Codebase

- [ ] My code follows the [Effective Go](https://go.dev/doc/effective_go) style guidelines.
- [ ] The code was planned and designed following the defined architecture and the separation of concerns.
- [ ] The code has sufficient comments, particularly for all hard-to-understand areas.
- [x] This PR adds value and shows no feature creep.
- [ ] I have augmented the test suite that proves my fix is effective or that my feature works.
- [ ] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

File: test/testkit/kyma/istio/istio.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package istio
```
Comment (ameteiko) [inline test/testkit/kyma/istio/istio.go:1]: `//go:build e2e` is missing

File: test/integration/istio_access_logs_test.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+				g.Expect(resp).To(HaveHTTPBody(SatisfyAll(
+					ContainLogs(WithAttributeKeys(istio.AccessLogAttributeKeys...)))))
+			}, timeout, interval).Should(Succeed())
+		})
+
+	})
+})
+
+func makeIstioAccessLogsK8sObjects(mockNs, mockDeploymentName, sampleAppNs string) ([]client.Object, *mocks.URLProvider, string) {
```
Comment (ameteiko) [inline test/integration/istio_access_logs_test.go:95]: Can we reuse the `makeLogsTestK8sObjects` or some of the parts of it?

File: test/integration/suite_test.go [hunk 3]
```diff
@@ -1,3 +1,106 @@
...
+	testEnv             *envtest.Environment
+	telemetryK8sObjects []client.Object
+)
+
+func TestE2e(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2e Suite")
+}
+
+var _ = BeforeSuite(func() {
```
Comment (ameteiko) [inline test/integration/suite_test.go:55]: Can we share this code with the E2E Acceptance tests (the main e2e test suite)?

File: test/testkit/mocks/urlprovider.go [hunk 4]
```diff
@@ -6,6 +6,7 @@ type URLProvider struct {
 	metrics   string
 	pipelines map[int]map[string]string
 	otlpPush  string
+	metricPod string
```
Comment (ameteiko) [inline test/testkit/mocks/urlprovider.go:9]: metricPod -> metricPodURL

**Review by ameteiko — CHANGES_REQUESTED** [review]

**Comment by rakesh-garimella** [comment]
related prow job run: https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_telemetry-manager/315/rakesh-garimella_test_of_prowjob_pull-telemetry-manager-integra/1688471194837192704

Prow Job PR: https://github.com/kyma-project/test-infra/pull/8561

**Review by ameteiko — APPROVED** [review]

**Pull request merged**